code/modules/mafia/controller.dm

```diff
@@ -651,10 +702,33 @@
 					helper = role
 					break
 			helper.show_help(usr)
-	if(!user_role || user_role.game_status == MAFIA_DEAD)//ghosts, dead people?
+	if(!user_role)//just the dead
+		var/client/C = ui.user.client
+		switch(action)
+			if("mf_signup")
+				if(!SSticker.HasRoundStarted())
+					to_chat(usr, "<span class='warning'>Wait for the game to start!</span>")
+					return
+				if(GLOB.mafia_signup[C.ckey])
+					GLOB.mafia_signup -= C.ckey
+					to_chat(usr, "<span class='notice'>You unregister from Mafia.</span>")
+					return
+				else
+					GLOB.mafia_signup[C.ckey] = C
+					to_chat(usr, "<span class='notice'>You sign up for Mafia.</span>")
+				if(phase == MAFIA_PHASE_SETUP)
+					check_signups()
+					try_autostart()
+			if("mf_spectate")
+				if(C.ckey in spectators)
+					to_chat(usr, "<span class='notice'>You will no longer get messages from the game.</span>")
+					spectators -= C.ckey
+				else
+					to_chat(usr, "<span class='notice'>You will now get messages from the game.</span>")
+					spectators += C.ckey
+	if(user_role.game_status == MAFIA_DEAD)
 		return
-	var/self_voting = user_role == on_trial ? TRUE : FALSE //used to block people from voting themselves innocent or guilty
```

Comment:
Was this removed ?

Reply:
It straight up doesn't work right now. I should make it work before this is merged